be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -831,10 +831,16 @@ Status BaseBetaRowsetWriter::_create_file_writer(const std::string& path,
     return Status::OK();
 }
 
-Status BaseBetaRowsetWriter::create_file_writer(uint32_t segment_id,
-                                                io::FileWriterPtr& file_writer) {
-    auto path = _context.segment_path(segment_id);
-    return _create_file_writer(path, file_writer);
+Status BaseBetaRowsetWriter::create_file_writer(uint32_t segment_id, io::FileWriterPtr& file_writer,
+                                                FileType file_type) {
+    auto segment_path = _context.segment_path(segment_id);
+    if (file_type == FileType::INVERTED_INDEX_FILE) {
+        std::string prefix =
+                std::string {InvertedIndexDescriptor::get_index_file_path_prefix(segment_path)};
+        std::string index_path = InvertedIndexDescriptor::get_index_file_path_v2(prefix);
+        return _create_file_writer(index_path, file_writer);
+    }
+    return _create_file_writer(segment_path, file_writer);
```

Comment:
do strict check for each type and return error in else branch, since more type may be added in the fulture.